ground/src/main/java/com/google/android/ground/ui/datacollection/PhotoDataCollectionFragment.kt

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.datacollection
+
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.content.FileProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.ground.BR
+import com.google.android.ground.BuildConfig
+import com.google.android.ground.databinding.EditSubmissionBottomSheetBinding
+import com.google.android.ground.databinding.PhotoDataCollectionFragBinding
+import com.google.android.ground.model.task.Task
+import com.google.android.ground.repository.UserMediaRepository
+import com.google.android.ground.rx.RxAutoDispose.autoDisposable
+import com.google.android.ground.ui.common.AbstractFragment
+import com.google.android.ground.ui.editsubmission.AddPhotoDialogAdapter
+import com.google.android.ground.ui.editsubmission.AddPhotoDialogAdapter.PhotoStorageResource
+import com.google.android.ground.ui.editsubmission.PhotoResult
+import com.google.android.ground.ui.editsubmission.PhotoTaskViewModel
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import dagger.hilt.android.AndroidEntryPoint
+import io.reactivex.Completable
+import timber.log.Timber
+
+/** Fragment allowing the user to capture a photo to complete a task. */
+@AndroidEntryPoint
+class PhotoDataCollectionFragment
+constructor(
+  private val task: Task,
+  private val viewModel: PhotoTaskViewModel,
```

Comment:
Please sync name of task with owning fragment.

Reply:
Done!

Reply:
Sorry, I was referring to the class name. `task` is find as a variable name and in comments since the type is clear from context. Re`PhotoDataCollectionFragment` vs `PhotoTaskViewModel`: `PhotoTask{Fragment|ViewModel}` would be sufficient. Open to other suggestions.

Reply:
Gotcha, I updated all the Data Collection Task names to {TaskType}TaskFragment.